raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -321,6 +335,23 @@ private void updateListenersProgress(List<ListenerContext> listenerContexts, lon
         }
     }
 
+    private Optional<SnapshotReader<T>> oldestSnapshot() {
+        if (log.oldestSnapshotId().isPresent()) {
```

Comment:
nit: we can use `log.oldestSnapshotId().map(oldestSnapshotId -> {...})` to get rid of if `else`

Reply:
Thanks. Fixed!